plugins/glustershd/actor.go

```diff
@@ -13,75 +14,149 @@ import (
 
 var names = [...]string{"replicate", "afr"}
 
+var selfhealKeys = []string{"replicate.self-heal-daemon", "afr.self-heal-daemon"}
+
 const (
 	selfHealKey = "self-heal-daemon"
 )
 
 type shdActor struct{}
 
-func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, logger log.FieldLogger) error {
-	if key != selfHealKey {
+func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, volOp xlator.VolumeOpType, logger log.FieldLogger) error {
+
+	voltype := v.Type.String()
+	if voltype != "Replicate" && voltype != "Disperse" {
 		return nil
 	}
+
 	glustershDaemon, err := newGlustershd()
 	if err != nil {
 		return err
 	}
+	switch volOp {
+	case xlator.VolumeStart:
+		for _, key := range selfhealKeys {
+			if v, ok := v.Options[key]; ok && v == "off" {
```

Comment:
v here is used to store volinfo as well as value for a xlator option. Can we change one of them maybe, to avoid any kind of confusion ?